src/compiler/checker.ts

```diff
@@ -22707,7 +22754,7 @@ namespace ts {
                         spread = getSpreadType(spread, createJsxAttributesType(), attributes.symbol, objectFlags, /*readonly*/ false);
                         attributesTable = createSymbolTable();
                     }
-                    const exprType = checkExpressionCached(attributeDecl.expression, checkMode);
+                    const exprType = getReducedType(checkExpressionCached(attributeDecl.expression, checkMode));

```

Comment:
Do we want to reduce before caching to avoid trying to repeat the reduction?

Reply:
No, `getReducedType` itself also caches, so it is cheap the second time around.